compiler/rustc_hir_typeck/src/inherited.rs

```diff
@@ -136,6 +140,45 @@ impl<'tcx> Inherited<'tcx> {
         if obligation.has_escaping_bound_vars() {
             span_bug!(obligation.cause.span, "escaping bound vars in predicate {:?}", obligation);
         }
+
+        let infcx = &self.infcx;
```

Comment:
I'd prefer if this stays in a separate method.

Also, I think `Inherited` autoderefs to `InferCtxt`, so there's no reason to not just use `self` in place of `infcx.